packages/client3/components/design-system/button/link-button.tsx

```diff
@@ -0,0 +1,19 @@
+import NextLink, { LinkProps as NextLinkProps } from "next/link";
+import React, { forwardRef } from "react";
+
+import { Button } from "./button";
+
+type LinkButtonProps = Pick<NextLinkProps, "href"> &
+  React.ComponentProps<typeof Button>;
+
+export const LinkButton = forwardRef<HTMLButtonElement, LinkButtonProps>(
+  function LinkButton({ href, ...rest }, ref) {
+    return (
+      <NextLink passHref href={href}>
+        <a>
+          <Button {...rest} ref={ref} />
+        </a>
+      </NextLink>
+    );
+  }
+);
```

Comment:
I didn't quite understand the reason to this component, we already have a "Link" component..

Reply:
Initially, the goldfinch button had a link button within it to handle links. So I just extracted it out as a separate component to handle it easily. The purpose of this component is to handle use cases where we need a page redirection from a button and also the current link component doesn't support having children.

Thoughts?

Reply:
thats fine I guess